hello_helpers/src/hello_helpers/hello_misc.py

```diff
@@ -104,75 +108,249 @@ def get_robot_floor_pose_xya(tf2_buffer, floor_frame='odom'):
+        return trigger_result.done()
+    
+    def switch_to_position_mode(self):
+        trigger_request = Trigger.Request()
+        trigger_result = self.switch_to_position_mode_service.call_async(trigger_request)
+        while not trigger_result.done():
+            time.sleep(0.2)
+        self.node.get_logger().debug(f"{self.node_name}'s HelloNode.switch_to_position_mode: got message {trigger_result.done()}")
+        return trigger_result.done()
+    
+    def switch_to_navigation_mode(self):
+        trigger_request = Trigger.Request()
+        trigger_result = self.switch_to_navigation_mode_service.call_async(trigger_request)
+        while not trigger_result.done():
+            time.sleep(0.2)
+        self.node.get_logger().debug(f"{self.node_name}'s HelloNode.switch_to_navigation_mode: got message {trigger_result.done()}")
+        return trigger_result.done()
+
+    def get_logger(self):
+        return self.node.get_logger()
+    
+    def get_tool(self):
+        assert(self.tool is not None)
+        return self.tool
+    
     def main(self, node_name, node_topic_namespace, wait_for_first_pointcloud=True):
-        super().__init__(node_name)
+        rclpy.init()
+        self.node = rclpy.create_node(node_name)
         self.node_name = node_name
-        self.get_logger().info("{0} started".format(self.node_name))
+        self.node.get_logger().info("{0} started".format(self.node_name))
+        
+        executor = MultiThreadedExecutor()
+        executor.add_node(self.node)
+        
+        self.new_thread = threading.Thread(target=self.spin_thread, args=(executor, ), daemon=True)
+        self.new_thread.start()
 
-        self.trajectory_client = ActionClient(self, FollowJointTrajectory, '/stretch_controller/follow_joint_trajectory')
+        reentrant_cb = ReentrantCallbackGroup()
+
+        self.trajectory_client = ActionClient(self.node, FollowJointTrajectory, '/stretch_controller/follow_joint_trajectory', callback_group=reentrant_cb)
         server_reached = self.trajectory_client.wait_for_server(timeout_sec=60.0)
         if not server_reached:
-            self.get_logger().error('Unable to connect to arm action server. Timeout exceeded.')
+            self.node.get_logger().error('Unable to connect to arm action server. Timeout exceeded.')
             sys.exit()
         
         self.tf2_buffer = tf2_ros.Buffer()
-        self.tf2_listener = tf2_ros.TransformListener(self.tf2_buffer, self)
+        self.tf2_listener = tf2_ros.TransformListener(self.tf2_buffer, self.node)
         
-        self.point_cloud_subscriber = self.create_subscription(PointCloud2, '/camera/depth/color/points', self.point_cloud_callback, 10)
-        self.point_cloud_pub = self.create_publisher(PointCloud2, '/' + node_topic_namespace + '/point_cloud2', 10)
+        self.tool_subscriber = self.node.create_subscription(String, '/tool', self.tool_callback, 10, callback_group=reentrant_cb)
```

Comment:
Do we have a /tool topic in Stretch Driver?

Reply:
Yes, we do!